### DISCUSSION TITLE
HMR and Hot hooks Issue -- Code is not updated

### DISCUSSION THREAD

**Discussion by salman3001** [adonisjs/core#4552] [Help] [answered]
Hi,

I was testing the new HMR and Hoot hooks feature. below are my settings.

```
//package.json
  "devDependencies": {
     ...
    "@adonisjs/assembler": "^7.5.1",
    "hot-hook": "^0.2.4",
    ...
  },
  "hotHook": {
    "boundaries": [
      "./app/controllers/**/*.ts",
      "./app/middleware/**/*.ts",
      "./database/**/*.ts",
      "./resources/*.ts"
    ]
  }
```

if i make changes to controller file, it works. vite does not refresh page and i get the updated code on next request. 

but the issue is if i make changes to services file which is injected as dependency in controller, vite still doesn't refresh the page that is good but the outdated code on executed on next request. i have to full reload server to get the updated code. 

not only with services file, but all the files except controller, like ploicies , validators etc.

but this was not the case with hothook v0.2.0. this is happening when i upgraded to v0.2.2. and now i also tried 0.2.4 its same.

**Comment by Julien-R44** [comment, marked as answer]
Why adding `"./database/**/*.ts"` and `"./resources/*.ts"` as boundaries? It's probably not what you want

If the problem persists after removing them, then a reproduction would be desirable please. Otherwise it will be hard to help you

**Reply by salman3001** [reply]
@Julien-R44  .. thanks for looking into this.  and it is working now. even keeping these options in boundaries. 

i dont know why it was not working before.  i ll let you know if i notice this issue again..

**Comment by kitro** [comment]
I have the same issue, I'm experimenting AdonisJs v6 with inertia adapter, I have a simple controller 

```ts
import User from '#models/user'
import type { HttpContext } from '@adonisjs/core/http'

export default class AboutController {
  async handle({ inertia }: HttpContext) {
    const users = await User.all()

    return inertia.render('about', { users: users, ok: 'test' })
  }
}
```

I'm running hmr with `npm run dev`  alias of `node ace serve --hmr` when i do the changes to `ok: 'test'` props it doesn't reflect in the browser, but in the console of `npm run dev` I got 
```
 INFO (50399): started HTTP server on localhost:3333
invalidated app/controllers/about/about_controller.ts (this log after I click crtl+s to save the file)
invalidated app/controllers/about/about_controller.ts (this log after I click crtl+s to save the file)
``` 
but the changes in `about_controller.ts` is correct, only after stop and run the `npm run dev` again I got the changes.

package.json for reference 
```
{
  "name": "adonis",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "license": "UNLICENSED",
  "scripts": {
    "start": "node bin/server.js",
    "build": "node ace build",
    "dev": "node ace serve --hmr",
    "test": "node ace test",
    "lint": "eslint .",
    "format": "prettier --write .",
    "typecheck": "tsc --noEmit"
  },
  "imports": {
    "#controllers/*": "./app/controllers/*.js",
    "#exceptions/*": "./app/exceptions/*.js",
    "#models/*": "./app/models/*.js",
    "#mails/*": "./app/mails/*.js",
    "#services/*": "./app/services/*.js",
    "#listeners/*": "./app/listeners/*.js",
    "#events/*": "./app/events/*.js",
    "#middleware/*": "./app/middleware/*.js",
    "#validators/*": "./app/validators/*.js",
    "#providers/*": "./providers/*.js",
    "#policies/*": "./app/policies/*.js",
    "#abilities/*": "./app/abilities/*.js",
    "#database/*": "./database/*.js",
    "#tests/*": "./tests/*.js",
    "#start/*": "./start/*.js",
    "#config/*": "./config/*.js"
  },
  "devDependencies": {
    "@adonisjs/assembler": "^7.5.2",
    "@adonisjs/eslint-config": "^1.3.0",
    "@adonisjs/prettier-config": "^1.3.0",
    "@adonisjs/tsconfig": "^1.3.0",
    "@japa/assert": "^3.0.0",
    "@japa/plugin-adonisjs": "^3.0.1",
    "@japa/runner": "^3.1.4",
    "@swc/core": "^1.5.3",
    "@types/luxon": "^3.4.2",
    "@types/node": "^20.12.10",
    "@vitejs/plugin-vue": "^5.0.4",
    "eslint": "^8.57.0",
    "hot-hook": "^0.2.0",
    "pino-pretty": "^11.0.0",
    "prettier": "^3.2.5",
    "ts-node": "^10.9.2",
    "typescript": "^5.4.5",
    "vite": "^5.2.11"
  },
  "dependencies": {
    "@adonisjs/auth": "^9.2.1",
    "@adonisjs/core": "^6.9.0",
    "@adonisjs/cors": "^2.2.1",
    "@adonisjs/inertia": "1.0.0-27",
    "@adonisjs/lucid": "^20.6.0",
    "@adonisjs/session": "^7.4.0",
    "@adonisjs/shield": "^8.1.1",
    "@adonisjs/static": "^1.1.1",
    "@adonisjs/vite": "^3.0.0-11",
    "@inertiajs/vue3": "^1.0.16",
    "@vinejs/vine": "^2.0.0",
    "@vue/server-renderer": "^3.4.27",
    "better-sqlite3": "^9.6.0",
    "edge.js": "^6.0.2",
    "luxon": "^3.4.4",
    "reflect-metadata": "^0.2.2",
    "vue": "^3.4.27"
  },
  "hotHook": {
    "boundaries": [
      "./app/controllers/**/*.ts",
      "./app/middleware/*.ts"
    ]
  },
  "eslintConfig": {
    "extends": "@adonisjs/eslint-config/app"
  },
  "prettier": "@adonisjs/prettier-config"
}
```

not sure where the issue was.

**Reply by Julien-R44** [reply]
Make sure you are using [lazy loading](https://docs.adonisjs.com/guides/basics/controllers#lazy-loading-controllers) for your controller.

If you are already using lazy loading, then I will need a minimal reproduction to be able to help you. Without reproduction I can't do anything here

**Reply by kitro** [reply]
Thank you Julien for your response. I tested with lazy loading controller still the issue persist until I refresh the page to get the changes.

this is a reproduction repo https://github.com/kitro/ado-test 

node version used `v21.7.3` 

when access to the route `http://localhost:3333/about`.  in the controller `app/controllers/about_controller.ts` I tried to update the `{ version: 12 }` props for testing if I can see the changes with hmr but is not working. I got this log in console `invalidated app/controllers/about_controller.ts`

**Reply by kitro** [reply]
This is a quick video show the issue

https://github.com/adonisjs/core/assets/5256921/80e06906-4d40-42fe-9847-5486613f60a5

**Reply by RomainLanz** [reply]
Your backend code is hot-reloaded correctly since the server doesn't have to restart to change the value.

What you are looking for is HMR on the client. Ensure your Vite configuration points to the correct assets.

**Reply by kitro** [reply]
I got you 👍 , I was thinking since Adonis is based on javascript I can get the hot-reloading also in the backend. my bad.
Thank you @RomainLanz 🙏 ,
